src/python/pants/base/deprecated.py

```diff
@@ -110,6 +116,9 @@ def warn_or_error(removal_version, deprecated_entity_description, hint=None, sta
   :param string deprecated_entity_description: A short description of the deprecated entity, that
                                             we can embed in warning/error messages.
   :param string hint: A message describing how to migrate from the removed entity.
+  :param string deprecation_start_version: The pantsbuild.pants version at which the entity will
```

Comment:
Document what the default means here.

Reply:
I've added:
> If not provided, the deprecation warning is always displayed.

Let me know if that's clear!

Reply:
SGTM